### PR TITLE
[BUGS#1164] Fix reloading TMX after team synchronization

### PR DESCRIPTION
**Pull request by briacp**
## Which ticket is resolved?

-  Translation memories are not correctly reloaded after team synchronization
  * https://sourceforge.net/p/omegat/bugs/1164

## What does this PR change?

- The TMX content refresh is placed inside the `saveAndRebase`callback to make sure all the TMX data is present.

### CODE REVIEW

File: src/org/omegat/core/data/RealProject.java [hunk 1]
```diff
@@ -1029,6 +1029,12 @@ public void parseHeadFile(File file) throws Exception {
                         public void rebaseAndSave(File out) throws Exception {
                             mergeTMX(baseTMX, headTMX, commitDetails);
                             projectTMX.exportTMX(config, out, false, false, true);
+
+                            // it can be not loaded yet
+                            ProjectTMX newTMX = new ProjectTMX(config.getSourceLanguage(), config.getTargetLanguage(),
+                                    config.isSentenceSegmentingEnabled(),
+                                    new File(config.getProjectInternalDir(), OConsts.STATUS_EXTENSION), null);
+                            projectTMX.replaceContent(newTMX);
```
Comment (miurahr) [inline src/org/omegat/core/data/RealProject.java:1037]: Is it better to be called after `ProjectTMX#replaceContent`?
```
 projectTMX.exportTMX(config, out, false, false, true);
```

Comment (miurahr) [reply src/org/omegat/core/data/RealProject.java:1037]: current implementation export TMX files before rebase and commit. Is it intended behavior?

File: src/org/omegat/core/data/RealProject.java [hunk 2]
```diff
@@ -1031,6 +1031,12 @@ public void parseHeadFile(File file) throws Exception {
...
```
[the remark this answers is not in the document]
Comment (miurahr) [reply src/org/omegat/core/data/RealProject.java:1039]: Yes, I have a question about it.

**Review by miurahr — APPROVED** [review]

**Comment by miurahr** [comment]
Last weekly release CI/CD failed with failure of integration-test.
When I bisect changes from the change here is a first commit to be bad.

A Ci/CD log here.
https://dev.azure.com/omegat-org/OmegaT/_build/results?buildId=6111&view=logs&j=f763054d-d222-5251-deca-727ca9ca5c85&t=cfd68ebe-c66e-5abf-ba3b-d83322e291ea

@briacp 

You can reproduce the problem using `docker` ;

```bash
git checkout 4e00736372fc1a88fb35920136bc754302cb4e64
env DURATION=300 TYPE=GIT docker-compose  up --abort-on-container-exit --exit-code-from client 
```

That will produce a log like

```
client_1  | 58347: Info: Reading of TMX file complete (TMXR_INFO_READING_COMPLETE)
client_1  | 26506: FINE: rebase and save 'omegat/project_save.tmx' 
client_1  | 26506: Info: Base:   [concurrent: 1688865263871, s2/0: 1] 
client_1  | 26506: Info: Mine:   [concurrent: 1688865263871, s2/0: 1, s2/1: 2] 
client_1  | 26506: Info: Theirs: [concurrent: 1688865269657, s1/0: 1] 
client_1  | 26506: Info: 'Theirs' TM is not a valid derivative of 'Base' TM 
```

When running on a good commit which is before merge here,  it will be like

```
client_1  | 37752: Info: Reading of TMX file complete (TMXR_INFO_READING_COMPLETE)
client_1  | 37752: FINE: rebase and save 'omegat/project_save.tmx' 
client_1  | 37752: Info: Base:   [concurrent: 1688860842643, s1/0: 1, s2/0: 1, s3/0: 1] 
client_1  | 37752: Info: Mine:   [concurrent: 1688860842643, s1/0: 1, s2/0: 1, s3/0: 1, s3/1: 2] 
client_1  | 37752: Info: Theirs: [concurrent: 1688860861554, s1/0: 1, s2/0: 3, s2/1: 2, s3/0: 1] 
client_1  | 37752: Info: Merged: [concurrent: 1688860861554, s1/0: 1, s2/0: 3, s2/1: 2, s3/0: 1, s3/1: 2] 
client_1  | 37752: Info: Git 'addForCommit' execution start (GIT_START)
client_1  | 37752: Info: Git 'addForCommit' execution finished successfully (GIT_FINISH)
client_1  | 37752: Info: Git 'upload' execution start (GIT_START)
```


As you know, a log message "rebase and save ..." is on `RebaseAndCommit` class Line 165
```java
        if (fileChangedLocally && fileChangedRemotely) {
            // rebase need only in case file was changed locally AND remotely
            Log.logDebug(LOGGER, "rebase and save '" + path + "'");
            needBackup = true;
            rebaser.rebaseAndSave(tempOut);
```
then `rebaseAndSave` method is one  the commit 4e0073637 chages.


IntegrationTest code `IntegrationTestChild#TestRealProject` class has a override method

```java
        @Override
        protected void mergeTMX(ProjectTMX baseTMX, ProjectTMX headTMX, StringBuilder commitDetails) {
            Log.log("Base:   " + baseTMX);
            Log.log("Mine:   " + projectTMX);
            Log.log("Theirs: " + headTMX);
           if (!checkMergeInput(baseTMX, projectTMX)) {
                Log.log("'Mine' TM is not a valid derivative of 'Base' TM");
```

that show a log message and check TMX inputs, the method is called from method which the change here modified.

```java

                        @Override
                        public void rebaseAndSave(File out) throws Exception {
                            mergeTMX(baseTMX, headTMX, commitDetails);

                            ProjectTMX newTMX = new ProjectTMX(config.getSourceLanguage(), config.getTargetLanguage(),
                                    config.isSentenceSegmentingEnabled(),
                                    new File(config.getProjectInternalDir(), OConsts.STATUS_EXTENSION), null);
                            projectTMX.replaceContent(newTMX);
```

It may be a senario that previous  merge, rebase, save and reload become wrong state.

**Comment by brandelune** [comment]
Please:
- add a line to changes.txt

**Pull request merged**